File: BudgetApp.Client/src/app/entities/Address.ts

export class Address {
  addressId: string;
  street: string;
  city: string;
  state: string;
  zipCode: number;

  constructor(addressId: string, street: string, city: string, state: string, zipCode: number) {
    this.addressId = addressId;
    this.street = street;
    this.city = city;
    this.state = state;
    this.zipCode = zipCode;
  }
}

File: BudgetApp.Domain/Entities/Address.cs
namespace BudgetApp.Domain.Entities;

public class Address
{
    public Guid AddressId { get; set; }
    public string Street { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string State { get; set; } = string.Empty;
    public int ZipCode { get; set; }
}